contracts/axelarnet-gateway/src/lib.rs

```diff
@@ -1,7 +1,7 @@
 pub mod contract;
 pub mod events;
 pub mod msg;
-pub mod state;
+mod state;
```

Comment:
Do we want to re-export state::Error publicly?

Reply:
I don't think it's necessary

Reply:
Maybe just for tests? I don't think other crates should use them, but in tests it is very helpful to ensure the correct error in a type-safe manner

Reply:
Ok, it's not needed for unit tests within the same crate. And due to the limitation of Rust, exporting it to others will require a `test` feature flag. We can add that when needed (mainly for integration tests)

Reply:
Ok, it's a bit ugly because of the same limitation of Rust which requires explicit feature gating for exposing test only methods.